grpc/creds/creds.go

```diff
@@ -51,25 +54,153 @@ func (tc *transportCredentials) ClientHandshake(ctx context.Context, addr string
 	}
 }
 
-// ServerHandshake performs the TLS handshake for a server <- client connection
-func (tc *transportCredentials) ServerHandshake(rawConn net.Conn) (net.Conn, credentials.AuthInfo, error) {
-	return nil, nil, fmt.Errorf("boulder/grpc/creds: Server-side handshakes are not implemented")
+// ServerHandshake is not implemented for a `clientTransportCredentials`, use
+// a `serverTransportCredentials` if you require `ServerHandshake`.
+func (tc *clientTransportCredentials) ServerHandshake(rawConn net.Conn) (net.Conn, credentials.AuthInfo, error) {
+	return nil, nil, fmt.Errorf(
```

Comment:
`fmt.Errorf` here is unnecessary (and a few places below), instead could you create a set of package level `error`s using `errors.New`. This should make testing the correct errors were returned a bit easier.